src/eq_schema/Questionnaire.js

```diff
@@ -15,36 +14,25 @@ class Questionnaire {
     this.data_version = "0.0.2";
     this.survey_id = questionnaireJson.surveyId || questionnaireId;
     this.title = questionnaireJson.title;
-    this.groups = this.buildGroups(questionnaireJson.sections);
+    this.sections = this.buildSections(questionnaireJson.sections);
     this.theme = questionnaireJson.theme;
     this.legal_basis = questionnaireJson.legalBasis;
-    this.navigation = this.buildNavigation(
-      questionnaireJson.navigation,
-      questionnaireJson.sections
-    );
+    this.navigation = this.buildNavigation(questionnaireJson.navigation);
 
     this.buildSummaryOrConfirmation(questionnaireJson.summary);
   }
 
-  buildGroups(sections) {
-    return sections.map(section => new Group(section));
+  buildSections(sections) {
+    return sections.map(section => new Section(section));
   }
 
   buildSummaryOrConfirmation(summary) {
     const finalPage = summary ? new Summary() : new Confirmation();
-    last(this.groups).blocks.push(finalPage);
+    last(this.sections).groups[0].blocks.push(finalPage);
```

Comment:
it should be in the last group of the last section not the first group of the last section?

Reply:
Does the last section contain a single group specifically for Summary/Confirmation page? If not then the logic likely needs to be `last(last(this.sections).groups)` as Jon mentions.

Reply:
I think the main issue might be that the final section is treated specially by runner. It either contains a summary or a confirmation and there's an assumption that it'll only contain a single group.

So whether we use the first or last group within the final section, there should only ever be a single group anyway.

Possibly something for schema-validator to handle.